core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -599,10 +598,9 @@ private object RTS {
                       val finalization = dispatchErrors(finalizer)
                       val completer    = io
 
-                      // Do not interrupt finalization:
-                      this.noInterrupt += 1
-
-                      curIo = finalization.widenError[E] *> completer
+                      curIo = enterUninterruptible *> finalization
```

Comment:
These refactorings will slightly decrease performance, but do increase clarity of the code and only happen in the finalizer code path, which is less common.

Reply:
Yeah, I thought the important part was to ensure `exitUninterruptible` ran, whereas `enterUninterruptible` is inoffensive. As this pattern appears in several places in the code, would it make sense to define
```scala
def doNotInterrupt[E](io: IO[E, Any]): IO[E, Any] = {
  this.noInterrupt += 1
  io.ensuring(exitUninterruptible)
}
```